plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -862,6 +865,197 @@ public void testThatEnableCdfTablePropertyIsShownForCtasTables()
+    {
+        String tableName = "test_create_table_with_cdf_enabled_and_unsupported_writer_version_" + randomNameSuffix();
+
+        assertQueryFails("CREATE TABLE " + tableName + " (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 3)",
+                "writer_version cannot be set less than 4 when cdf is enabled");
+        assertQueryFails("CREATE TABLE " + tableName + " (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 5)",
+                "Unable to set catalog 'delta_lake' table property 'writer_version' to \\[5]: writer_version must be between 2 and 4");
+        assertQueryFails("CREATE TABLE " + tableName + " (a_number) WITH (change_data_feed_enabled = true, writer_version = 3) AS VALUES (1), (2)",
+                "writer_version cannot be set less than 4 when cdf is enabled");
+        assertQueryFails("CREATE TABLE " + tableName + " (a_number) WITH (change_data_feed_enabled = true, writer_version = 5) AS VALUES (1), (2)",
+                "Unable to set catalog 'delta_lake' table property 'writer_version' to \\[5]: writer_version must be between 2 and 4");
+    }
+
+    @Test
+    public void testAlterTableWithInvalidReaderWriterVersion()
+    {
+        String tableName = "test_alter_table_with_invalid_reader_writer_version_" + randomNameSuffix();
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT)", SCHEMA, tableName));
+
+        assertQueryFails("ALTER TABLE " + tableName + " SET PROPERTIES reader_version = 0",
+                "Unable to set catalog 'delta_lake' table property 'reader_version' to \\[0]: reader_version must be between 1 and 2");
+        assertQueryFails("ALTER TABLE " + tableName + " SET PROPERTIES reader_version = 3",
+                "Unable to set catalog 'delta_lake' table property 'reader_version' to \\[3]: reader_version must be between 1 and 2");
+        assertQueryFails("ALTER TABLE " + tableName + " SET PROPERTIES writer_version = 1",
+                "Unable to set catalog 'delta_lake' table property 'writer_version' to \\[1]: writer_version must be between 2 and 4");
+        assertQueryFails("ALTER TABLE " + tableName + " SET PROPERTIES writer_version = 5",
+                "Unable to set catalog 'delta_lake' table property 'writer_version' to \\[5]: writer_version must be between 2 and 4");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testAlterTableUpgradeReaderWriterVersion()
+    {
+        String tableName = "test_alter_table_upgrade_reader_writer_version_" + randomNameSuffix();
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT)", SCHEMA, tableName));
+
+        assertUpdate("ALTER TABLE " + tableName + " SET PROPERTIES reader_version = 2");
+        assertThatShowCreateTable(tableName, ".*reader_version = 2.*");
+
+        assertUpdate("ALTER TABLE " + tableName + " SET PROPERTIES writer_version = 3");
+        assertThatShowCreateTable(tableName, ".*writer_version = 3.*");
```

Comment:
Can we also somehow check if transaction log is updated with new entry?

Reply:
`SHOW CREATE TABLE ...` actually does the same thing. It reads the transaction log file, parses and sets the `protocolEntry` and other objects. So I think it is safe to assume that the version returned from `SHOW CREATE TABLE` is the same as of transaction log.

https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TransactionLogTail.java#L138